### PR TITLE
feat: add option to create private-only basic hpcs

### PR DESCRIPTION
**Pull request by Aashiq-J**
### Description

Option to create private-only basic hpcs instance.

#### Changes that affect the core Terraform module or submodules
- [ ] Bug fix
- [x] New feature
- [ ] Dependency update

#### Changes that don't affect the core Terraform module or submodules
- [ ] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other

#### Release required?
Identify the type of release. For information about the changes in a semantic versioning release, see [Release versioning](https://terraform-ibm-modules.github.io/documentation/#/versioning).

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

Support for creating private-only basic hpcs.

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author. The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).
- Merge by using "Squash and merge".

### CODE REVIEW

**Comment by Aashiq-J** [comment]
For the argument reference `allowed_network` I have kept the input as `var.service_endpoints` because the variable name remains consistent across the two approaches of hpcs creation.

File: examples/basic/variables.tf [hunk 1]
```diff
@@ -27,3 +27,9 @@ variable "resource_tags" {
   description = "Optional list of tags to apply to the HPCS instance."
   default     = []
 }
+
+variable "service_endpoints" {
+  type        = string
+  description = "The service_endpoints to access your service instance."
+  default     = "private-only"
```
Comment (maheshwarishikha) [inline examples/basic/variables.tf]: I think ..default value should be public-and-private

File: outputs.tf [hunk 2]
```diff
@@ -22,4 +22,9 @@ output "crn" {
   value       = var.auto_initialization_using_recovery_crypto_units ? ibm_hpcs.hpcs_instance[0].crn : ibm_resource_instance.base_hpcs_instance[0].crn
 }
 
+output "extensions" {
```
Comment (maheshwarishikha) [inline outputs.tf]: can we say endpoints instead of extensions?

File: outputs.tf [hunk 3]
```diff
@@ -22,4 +22,9 @@ output "crn" {
   value       = var.auto_initialization_using_recovery_crypto_units ? ibm_hpcs.hpcs_instance[0].crn : ibm_resource_instance.base_hpcs_instance[0].crn
 }
 
+output "endpoints" {
+  description = "HPCS instance extensions"
```
Comment (maheshwarishikha) [inline outputs.tf]: We are nowhere using the term `extensions` except this. It is also reflected in ReadMe too. So better to change it as `endpoints` to maintain consistency.

**Comment by vburckhardt** [comment]
Thanks for the PR - could we check all permutation of this parameter for both the standard and hybrid hpcs?

**Comment by maheshwarishikha** [comment]
> Thanks for the PR - could we check all permutation of this parameter for both the standard and hybrid hpcs?

I'll test for hybrid-HPCS today.

**Comment by maheshwarishikha** [comment]
Tested with Hybrid HPCS. Instance was created with `private-only` endpoint.

image

**Review by maheshwarishikha — APPROVED** [review]

**Comment by vburckhardt** [comment]
One addition here would be to add the service endpoint variable in the hybrid example.

**Pull request merged**